.github/actions/calc_vars/action.yml

```diff
@@ -0,0 +1,47 @@
+name: calc vars
+description: calculate variables based off the target name that may be useful in other steps
+
+inputs:
+  target:
+    description: 'target architecture'
+
+outputs:
+  arch:
+    description: "base architecture - one of x86, x86_64, arm, aarch64 or riscv64"
+    value: ${{ steps.calc_vars_action.outputs.arch }}
+  toolchain:
+    description: "path to toolchain file for architecture"
+    value: ${{ steps.calc_vars_action.outputs.toolchain }}
+  cmake_toolchain:
+    description: "cmake argument to pass to "    
+    value: ${{ steps.calc_vars_action.outputs.cmake_toolchain }}    
+runs:
+  # We don't want a new docker just a list of steps, so mark as composite
+  using: "composite"
+  steps:
+    - shell: bash
+      id: calc_vars_action
+      run: |
+        export ARCH=
```

Comment:
All the `export`s here don't do anything, since you are not running any processes that look at environment variables. Regular shell variables will work just as well.